src/combinator.ts

```diff
@@ -107,6 +108,8 @@ export const or4P = <A, B, C, D>(
   return or2P(or2P(first, second), or2P(third, fourth));
 };
 
+export const optionalP = <T>(proxy: TypeProxy<T>) => or2P(undefinedP, proxy);
```

Comment:
I think the variable name `proxy` should be `type` right? To match the other functions

Reply:
yeah, updated.